Marlin/src/HAL/HAL_LPC1768/HardwareSerial.h

```diff
@@ -28,54 +28,41 @@
 #include <Stream.h>
 
 extern "C" {
-  #include <debug_frmwrk.h>
-  //#include <lpc17xx_uart.h>
+  #include <lpc17xx_uart.h>
+  #include "lpc17xx_pinsel.h"
 }
 
-#define IER_RBR   0x01
-#define IER_THRE  0x02
-#define IER_RLS   0x04
-
-#define IIR_PEND  0x01
-#define IIR_RLS   0x03
-#define IIR_RDA   0x02
-#define IIR_CTI   0x06
-#define IIR_THRE  0x01
-
-#define LSR_RDR   0x01
-#define LSR_OE    0x02
-#define LSR_PE    0x04
-#define LSR_FE    0x08
-#define LSR_BI    0x10
-#define LSR_THRE  0x20
-#define LSR_TEMT  0x40
-#define LSR_RXFE  0x80
-
-#define UARTRXQUEUESIZE   0x10
-
 class HardwareSerial : public Stream {
 private:
-uint8_t PortNum;
-uint32_t baudrate;
+  static const size_t UARTRXQUEUESIZE = 256;
+
```

Comment:
https://github.com/MarlinFirmware/Marlin/blob/f6f881f5ed954bc2f44ca64895b6af99e74265a2/Marlin/Configuration_adv.h#L742-L746

Reply:
Having no TX-buffer nods also a comment in Configuration_adv.h

Reply:
I didn't realize those were defined in Configuration_adv.h. I'm working on getting them included and I'll also add support for buffered transmissions. I suppose we'll need to add XON/XOFF support at some point, but I'd like to get the basic serial implementation going first. 

From what I can tell, `RX_BUFFER_SIZE` can't actually be 0. Would you agree? If so, I'll update the comment in the Configuration_adv.h files as part of the change.